instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -0,0 +1,116 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO: usage
+"""
+The opentelemetry-instrumentation-aws-lambda package allows tracing AWS
+Lambda function.
+
+Usage
+-----
+
+.. code:: python
+    # Create a Python function in AWS Lambda console
+    # Ref Doc: https://docs.aws.amazon.com/lambda/latest/dg/lambda-python.html
+
+    import boto3
+    from opentelemetry.instrumentation.aws_lambda import (
+        AwsLambdaInstrumentor
+    )
+
+    # Enable instrumentation
+    AwsLambdaInstrumentor().instrument()
+
+    # Lambda function
+    def lambda_handler(event, context):
+        s3 = boto3.resource('s3')
+        for bucket in s3.buckets.all():
+            print(bucket.name)
+
+        return "200 OK"
+
+API
+---
+"""
+
+import logging
+import os
+from importlib import import_module
+
+from wrapt import wrap_function_wrapper
+
+from opentelemetry.instrumentation.aws_lambda.version import __version__
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import unwrap
+from opentelemetry.sdk.extension.aws.trace.propagation.aws_xray_format import (
+    TRACE_HEADER_KEY,
+    AwsXRayFormat,
+)
+from opentelemetry.trace import SpanKind, get_tracer, get_tracer_provider
+from opentelemetry.trace.propagation.textmap import DictGetter
+
+logger = logging.getLogger(__name__)
+
+
+class AwsLambdaInstrumentor(BaseInstrumentor):
+    # pylint: disable=attribute-defined-outside-init
+    def _instrument(self, **kwargs):
+        self._tracer = get_tracer(
+            __name__, __version__, kwargs.get("tracer_provider")
+        )
+
+        self._tracer_provider = get_tracer_provider()
+
+        lambda_handler = os.environ.get("_HANDLER")
+        wrapped_names = lambda_handler.rsplit(".", 1)
+        self._wrapped_module_name = wrapped_names[0]
+        self._wrapped_function_name = wrapped_names[1]
+
+        wrap_function_wrapper(
+            self._wrapped_module_name,
+            self._wrapped_function_name,
+            self._function_patch,
+        )
+
+    def _uninstrument(self, **kwargs):
+        unwrap(
+            import_module(self._wrapped_module_name),
+            self._wrapped_function_name,
+        )
+
+    def _function_patch(self, original_func, _, args, kwargs):
+        lambda_context = args[1]
+        ctx_aws_request_id = lambda_context.aws_request_id
+        orig_handler = os.environ.get(
+            "ORIG_HANDLER", os.environ.get("_HANDLER")
+        )
+        xray_trace_id = os.environ.get("_X_AMZN_TRACE_ID", "")
+
+        propagator = AwsXRayFormat()
+        parent_context = propagator.extract(
+            DictGetter(), {TRACE_HEADER_KEY: xray_trace_id}
+        )
+
+        with self._tracer.start_as_current_span(
+            orig_handler, context=parent_context, kind=SpanKind.CLIENT
+        ) as span:
```

Comment:
should this be checking if span `is_recording` is true before setting attributes? also looking at the spec there appears to be more attributes that are needed in order to adhere to the spec